x-pack/plugins/reporting/public/components/report_info_button.tsx

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonIcon, EuiCallOut, EuiPopover } from '@elastic/eui';
+import { get } from 'lodash';
+import React, { Component, Fragment } from 'react';
+import { JobInfo, jobQueueClient } from '../lib/job_queue_client';
+
+interface Props {
+  jobId: string;
+}
+
+interface State {
+  isLoading: boolean;
+  isPopoverOpen: boolean;
+  calloutTitle: string;
+  info: JobInfo | null;
+}
+
+export class ReportInfoButton extends Component<Props, State> {
+  private mounted?: boolean;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isLoading: false,
+      isPopoverOpen: false,
+      calloutTitle: 'Job info',
+      info: null,
+    };
+  }
+
+  public renderInfo() {
+    const { info } = this.state;
+    if (!info) {
+      return null;
+    }
+
+    const iGet = (path: string) => get(info, path, 'n/a');
```

Comment:
Small nit: this helper gets made everytime `renderInfo` is called. We could probably extract this around line 22 so the browser can do whatever optimizations it can in its JIT (likely small).

Reply:
True, since it was enclosing `info`. I changed it to just use lodash directly from the JSX